.golangci.yml

```diff
@@ -6,21 +6,50 @@ run:
 
 linters:
   enable:
-    - protogetter # reports direct reads from proto message fields when getters should be used.
-  # Right now revive raises 50+ issues, whereas golint didn't have any.
-  # despite revive being recommended as a replacement for golint.
-  # TODO: should we turn on revive and fix the issues?
-  #     - revive # drop-in replacement for golint
-  exclusions:
+    - protogetter # reports direct reads from proto message fields when getters should be used
+    - revive      # drop-in replacement for golint
+    - lll         # enforces line length limits
+    - errorlint   # makes sure errors are wrapped correctly
+    - misspell    # checks for common misspellings
+    - funlen      # limits function length
+    - nestif      # limits nesting depth
+    - exhaustive  # makes sure enum switch statements are exhaustive
+    - errcheck    # enforces that all errors are checked
+    - unused      # checks for unused constants, variables, functions and types
+    - unconvert   # removes unnecessary type conversions
+    - wrapcheck   # checks that errors returned from external packages are wrapped
+    - exhaustruct # checks if all structure fields are initialized
```

Comment:
this I think can be annoying, and also goes against the golang philosophy of not initializing default values?

Reply:
> One possible counter-argument is if a config struct has a lot of useful default zero values

Fair point. I think in this case, I still prefer adding test overhead to gain the added safety. There's always the option of making a "default config", which in practice sidesteps this linting rule, and lets you rely on default values.